### ISSUE TITLE
Email speakers when videos are published

### ISSUE THREAD

**Issue by russss** [emfcamp/Website#1259]
It would be good to email speakers a link to their talk video when it's published. Some thoughts:

* We should probably abstract this somewhere, so we can set talk videos through an internal API or through a command-line tool. (We have some existing video-matching code [here](https://github.com/emfcamp/Website/blob/main/apps/base/tasks_videos.py) but it's a bit brittle).
* We should only email the speaker the first time this happens (videos may not be uploaded to all video services simultaneously).
* We should probably also email speakers when the "video lost" flag is set (#1258)

**Comment by jayaddison** [comment]
Some ad-hoc requirements gathering from IRC:

- [ ] On the EMF talk/proposal view ([example](https://www.emfcamp.org/schedule/2022/139-a-brief-history-of-time-zones-and-daylight-saving-time)), add a list of hyperlinks to video URLs below the embedded video object.
- [ ] When any non-empty URL is first configured for a `Proposal`, send a one-time email to the speaker with a link to the EMF talk/proposal view URL.

**Comment by fkusei** [comment]
I'm currently working on a "generic webhook" feature for our publishing scripts which would maybe help here: https://github.com/voc/voctopublish/pull/109 (feedback welcome)

(Requires changing the API on the website though)

**Comment by wlcx** [comment]
I'm not 100% on emailing the speaker in the case of missing recording - there may be more nuance to it than a generic catch all "something went wrong" message, and in any case I think I'd rather email them with a more personal message and a name attached rather than a "cold" website email.

Of course, that's just me and it's probably better they get something rather than nothing - so maybe the best course is to proceed but also send them a more human email? Just thinking out loud.

**Comment by crablab** [comment]
Agreed. I don't think we should automatically email people with lost recordings. 

We should probably send an alert to the CfP channel instead, so someone from the Content team can reach out directly and discuss with the speaker.

That assumes this is a relatively rare occurrence?

**Comment by russss** [comment]
Historically it has not really been a rare occurrence but I'm hopeful we won't have any lost recordings this year, so I'm happy to skip that for the moment at least.

**Comment by russss** [comment]
Some goalpost-moving, sorry:

* It looks like we will need a way of sending the emails after the URLs are set. I wonder if it would be best to have this as a command-line task, but we'd need to store the email status somewhere. We could maybe add a "video-published" state to the state machine?
* We may not make YouTube videos public at the same time that they're uploaded. So I think it's best to make this trigger on the `c3voc_url` only for the moment.

**Comment by Jonty** [comment]
> * We may not make YouTube videos public at the same time that they're uploaded. So I think it's best to make this trigger on the `c3voc_url` only for the moment.

If we do this, we should make sure that we include copy in the email telling them that the video will also be published on YouTube at a later date, just so they know this is happening.

**Comment by fkusei** [comment]
In its current implementation, the c3voc webhook will set the youtube url as well, even if the youtube videos are set to private. (There is currently no indication on whether the videos are publicly available, but we can implement that)

**Comment by russss** [comment]
The main considerations here are:

1. Youtube may refuse to publish a video after upload due to ContentID detection or whatever
2. It's nicer for people who subscribe to the EMF Youtube channel if all the videos don't get dumped into their feed in quick succession. (Particularly if they happen to be attending EMF at the time.)
3. It's better for engagement if we publish no more than 3 videos per day because the videos get de-prioritised by youtube after that. (I'm not too concerned about this one, but it does tie in with point 2)

I wonder if we can easily check if the youtube video is public when we send the email? Or c3voc could do that and send us an updated webhook.

Or we just omit the youtube URL from the email and say "Your video will be published on youtube soon", but that still means we'll be linking to a private YT video from our talk pages.

**Comment by jayaddison** [comment]
> > * We may not make YouTube videos public at the same time that they're uploaded. So I think it's best to make this trigger on the `c3voc_url` only for the moment.
> 
> If we do this, we should make sure that we include copy in the email telling them that the video will also be published on YouTube at a later date, just so they know this is happening.

Would applying both of those suggestions in combination provide a complete-enough solution?

Gradual release of YouTube videos to channels would then (I think?) be isolated from the email-notification logic.

**Comment by jayaddison** [comment]
> Gradual release of YouTube videos to channels would then (I think?) be isolated from the email-notification logic.

Sorry, I missed one aspect of the requirements:

> Or we just omit the youtube URL from the email and say "Your video will be published on youtube soon", but that still means we'll be linking to a private YT video from our talk pages.

I don't like that I keep missing important details and adding distraction, so I'm going to step back from this.
